crates/wasmtime/src/runtime/vm/gc/enabled/data.rs

```diff
@@ -60,40 +60,46 @@ impl PodValType<{ mem::size_of::<V128>() }> for V128 {
 /// for preserving the memory safety of indexed GC heaps in the face of (for
 /// example) collector bugs, but the latter is just a defensive technique to
 /// catch bugs early and prevent action at a distance as much as possible.
-pub struct VMGcObjectDataMut<'a> {
-    data: &'a mut [u8],
+pub struct VMGcObjectData<T> {
+    data: T,
 }
```

Comment:
If you want to get really fancy one option is to define this as `VmGcObjectData([u8])` to avoid needing generics here at all.

Reply:
Then we need unsafe code to construct it tho, because we can't use the unbox feature, right?

Reply:
Right yeah, we'd have to have a transmute for that (but IMO is reasonable, nothing compared to the other unsafe bits we have)